src/bluesky/tests/test_cross_section_image.py

```diff
@@ -0,0 +1,248 @@
+from unittest import TestCase
+from unittest.mock import MagicMock, call, patch
+
+import numpy as np
+import pytest
+from matplotlib import colormaps
+from matplotlib import pyplot as plt
+from matplotlib.colors import (
+    Colormap,
+    Normalize,
+)
+from matplotlib.figure import Figure
+
+from bluesky.callbacks._mpl_image_cross_section import (
+    CrossSection,
+    InterpolationEnum,
+    auto_redraw,
+)
+from bluesky.callbacks.broker import LiveImage
+
+
+@pytest.fixture
+def mock_matplotlib():
+    with patch("matplotlib.pyplot.figure") as mock_fig:
+        mock_figure_instance = MagicMock(spec=plt.Figure)
+        mock_subplot = MagicMock(spec=plt.Axes)
+        mock_figure_instance.add_subplot.return_value = mock_subplot
+        mock_subplot.get_figure.return_value = mock_figure_instance
+        mock_fig.return_value = mock_figure_instance
+        yield mock_fig
+
+
+@pytest.fixture
+def mock_cross_section():
+    return CrossSection(figure=MagicMock())
+
+
+@pytest.fixture
+def live_image(mock_cross_section):
+    return LiveImage(
+        "test_field",
+        cmap=Colormap(colormaps["magma"]),
+        norm=Normalize(),
+        interpolation=InterpolationEnum.NONE,
+        cross_section=mock_cross_section,
+    )
+
+
+class TestAutoRedrawDecorator(TestCase):
+    def setUp(self):
+        class TestClass:
+            def __init__(self):
+                self._fig = MagicMock()
+                self._auto_redraw = True
+
+            @auto_redraw
+            def some_function(self, x, y):
+                return x + y
+
+            def _update_artists(self):
+                pass
+
+            def _draw(self):
+                pass
+
+        self.obj = TestClass()
+
+    def test_function_call_and_return_value(self):
+        result = self.obj.some_function(2, 3)
+        self.assertEqual(result, 5, "The decorated function should return the sum of inputs")
+
+    def test_redraw_called_when_auto_redraw_true(self, mock_cross_section: CrossSection):
+        mock_cross_section.draw_idle()
+        self.obj.some_function(2, 3)
+        self.obj._fig.canvas.assert_not_called()
+        self.obj._update_artists.assert_called_once()
+        self.obj._draw.assert_called_once()
```

Comment:
A cleaner way to mock this is:
```suggestion
@auto_redraw
def some_function(obj, x, y):
    return x + y

class TestAutoRedrawDecorator(TestCase):
    def test_function_call_and_return_value(self):
        result = some_function(MagicMock(), 2, 3)
        self.assertEqual(result, 5, "The decorated function should return the sum of inputs")

    def test_redraw_called_when_auto_redraw_true(self):
        mock_obj = MagicMock()
        mock_obj._auto_redraw = True
        some_function(mock_obj, 2, 3)
        mock_obj._fig.canvas.assert_not_called()
        mock_obj._update_artists.assert_called_once()
        mock_obj._draw.assert_called_once()
```

And so on...